### DISCUSSION TITLE
[React] more options in `useDismiss` hook (control key and close button)

### DISCUSSION THREAD

**Discussion by mayank99** [floating-ui/floating-ui#2721] [React]
**Is your feature request related to a problem? Please describe.**

The `useDismiss` hook has some great toggleable defaults - it dismisses when pressing <kbd>Esc</kbd> or when clicking outside.

However, to make tooltips more accessible, it is recommended to:
- dismiss the tooltip when pressing <kbd>ctrl</kbd> instead of <kbd>esc</kbd>
- provide a close button for mouse users

This recommendation comes from Sarah Higley, who tested and received feedback from real users. See video: https://youtu.be/lb0_v7D4kbs?t=1436

**Describe the solution you'd like**

1. Ability to allow changing the dismiss key.
```js
useDismiss(context, {
  escapeKey: false,
  customKey: 'Control',
  // or 
  customKey: e => e.ctrlKey
})
```

2. Ability to add a close button which can be positioned either inside the tooltip or over the "arrow".
```js
<FloatingArrow>
  <FloatingCloseButton>❌</FloatingCloseButton>
</FloatingArrow>
```

**Describe alternatives you've considered**

A close button is relatively easy to add manually, but wiring up the control key handlers takes a bit of extra code.

**Additional context**

Highly recommend watching the video linked above (timestamped) to understand the need for this.

**Comment by Riyaancode** [comment]
Hi @mayank99 & @atomiks can I work on this feature?

**Comment by lokeshvasnik** [comment]
I would like to contribute

**Comment by andreidoroshevich** [comment]
Hey, guys

When will this feature be implemented?

**Comment by andreidoroshevich** [comment]
Or maybe you can advice me something if I need to close menu with this library after it loses focus?
